codegen/src/component.rs

```diff
@@ -194,7 +194,7 @@ impl ComponentMeta {
 
             Some(quote! {
                 #[derive(Clone)]
-                struct #ident(std::rc::Rc<std::cell::RefCell<ruukh::component::Status<#state_ty>>>);
+                pub struct #ident(std::rc::Rc<std::cell::RefCell<ruukh::component::Status<#state_ty>>>);
```

Comment:
It seems you are making the status struct public by default. We should rather inherit the visibility from the component struct.